vcd/datasource_vcd_nsxt_security_group.go

```diff
@@ -63,34 +70,72 @@ func datasourceVcdNsxtSecurityGroup() *schema.Resource {
 func datasourceVcdSecurityGroupRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	vcdClient := meta.(*VCDClient)
 
-	nsxtEdgeGateway, err := vcdClient.GetNsxtEdgeGatewayFromResourceById(d, "edge_gateway_id")
+	org, err := vcdClient.GetOrgFromResource(d)
 	if err != nil {
-		return diag.Errorf(errorUnableToFindEdgeGateway, err)
+		return diag.Errorf("[nsxt security group read] error retrieving Org: %s", err)
 	}
 
-	// Name uniqueness is enforce by VCD for types.FirewallGroupTypeSecurityGroup
-	secGroup, err := nsxtEdgeGateway.GetNsxtFirewallGroupByName(d.Get("name").(string), types.FirewallGroupTypeSecurityGroup)
+	ipSetName := d.Get("name").(string)
```

Comment:
Or something similar
```suggestion
	secourityGroupName := d.Get("name").(string)
```

Reply:
fixed